permissions/src/main/java/com/google/accompanist/permissions/MutableMultiplePermissionsState.kt

```diff
@@ -0,0 +1,149 @@
+    // Refresh permissions when the lifecycle is resumed.
+    PermissionsLifecycleCheckerEffect(mutablePermissions)
+
+    val multiplePermissionsState = remember(permissions) {
+        MutableMultiplePermissionsState(mutablePermissions)
+    }
+
+    // Remember RequestMultiplePermissions launcher and assign it to multiplePermissionsState
+    val launcher = rememberLauncherForActivityResult(
+        ActivityResultContracts.RequestMultiplePermissions()
+    ) { permissionsResult ->
+        multiplePermissionsState.updatePermissionsStatus(permissionsResult)
+    }
+    DisposableEffect(multiplePermissionsState, launcher) {
+        multiplePermissionsState.launcher = launcher
+        onDispose {
+            multiplePermissionsState.launcher = null
+        }
+    }
+
+    return multiplePermissionsState
+}
+
+@Composable
+private fun rememberMutablePermissionsState(
+    permissions: List<String>
+): List<MutablePermissionState> {
+    // Create list of MutablePermissionState for each permission
+    val context = LocalContext.current
+    val mutablePermissions: List<MutablePermissionState> = remember(permissions) {
+        permissions.map { MutablePermissionState(it, context, context.findActivity()) }
+    }
+    // Update each permission with its own launcher
+    for (permissionState in mutablePermissions) {
+        key(permissionState.permission) {
+            // Remember launcher and assign it to the permissionState
+            val launcher = rememberLauncherForActivityResult(
+                ActivityResultContracts.RequestPermission()
+            ) {
+                permissionState.setHasPermission(it)
+            }
+            DisposableEffect(launcher) {
+                permissionState.launcher = launcher
+                onDispose {
+                    permissionState.launcher = null
+                }
+            }
+        }
+    }
+
+    return mutablePermissions
+}
+
+/**
+ * A state object that can be hoisted to control and observe multiple permission status changes.
+ *
+ * In most cases, this will be created via [rememberMutableMultiplePermissionsState].
+ *
+ * @param mutablePermissions list of mutable permissions to control and observe.
+ */
+@Stable
+internal class MutableMultiplePermissionsState(
+    private val mutablePermissions: List<MutablePermissionState>
+) : MultiplePermissionsState {
+
+    override val permissions: List<PermissionState> = mutablePermissions
+
+    override val revokedPermissions: List<PermissionState> by derivedStateOf {
+        permissions.filter { !it.hasPermission }
+    }
+
+    override val allPermissionsGranted: Boolean by derivedStateOf {
+        permissions.all { it.hasPermission } || // Up to date when the lifecycle is resumed
+            revokedPermissions.isEmpty() // Up to date when the user launches the action
+    }
+
+    override val shouldShowRationale: Boolean by derivedStateOf {
+        permissions.firstOrNull { it.shouldShowRationale } != null
```

Comment:
Nit: Feels like `any { it.shouldShowRationale != null }` is a better fit?

Reply:
It is! Thanks